config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -140,6 +144,49 @@ public void deleteConfig(
     }
   }
 
+  @Override
+  public void deleteConfigs(
+      DeleteConfigsRequest request, StreamObserver<DeleteConfigsResponse> responseObserver) {
+    try {
+      if (request.getConfigsCount() == 0) {
+        responseObserver.onError(
+            Status.INVALID_ARGUMENT
+                .withDescription("List of configs to delete provided is empty")
+                .asException());
+        return;
+      }
+      Map<ConfigResourceContext, Value> valuesByContext =
+          request.getConfigsList().stream()
+              .map(
+                  requestedDelete ->
+                      Map.entry(this.getConfigResourceContext(requestedDelete), emptyValue()))
+              .collect(ImmutableMap.toImmutableMap(Entry::getKey, Entry::getValue));
+
+      List<UpsertedConfig> deletedConfigs =
```

Comment:
Wouldn't this be returning the post-delete values (aka empty). The response from this API I would expect would be the values that have been deleted, so if we're not going to return that we should be modifying the API.

Reply:
I take it back - I see it returns the prev config too :)